assets/ajax-reload.js

```diff
@@ -0,0 +1,142 @@
+window.HasteAjaxReload = {
+    elementsInProgress: {},
+    eventsInProgress: {},
+
+    /**
+     * Dispatch the events
+     */
+    dispatchEvents: function () {
+        if (arguments.length < 1) {
+            console.error('Please provide at least one event');
+        }
+
+        var els = {};
+        var events = [];
+        var listeners = document.querySelectorAll('[data-haste-ajax-listeners]');
+
+        Array.from(arguments).forEach(function (event) {
+            var found = false;
+
+            // Find the elements that listen to particular event
+            Array.from(listeners).forEach(function (el) {
+                if (el.dataset.hasteAjaxListeners.split(' ').indexOf(event) !== -1) {
+                    found = true;
+                    els[el.dataset.hasteAjaxId] = el;
+                }
+            });
+
+            if (!found) {
+                console.warn('There are no eligible elements for event "' + event + '"');
```

Comment:
I don't think we should trigger a warning here. I mean, there's nothing wrong with that case :-)